reliability/single-fit/trace/src/test/kotlin/com/datadog/android/trace/integration/opentracing/SpanExt.kt

```diff
@@ -7,34 +7,31 @@
 package com.datadog.android.trace.integration.opentracing
 
 import com.datadog.android.core.internal.utils.toHexString
+import com.datadog.opentracing.DDSpan
 import io.opentracing.Span
 import io.opentracing.SpanContext
 
 /**
- * Returns the span's traceId in hex format.
- * The [SpanContext.toTraceId] method returns a string in decimal format,
- * which doesn't match what we send in our events
+ * Returns the span's least significant trace id in hex format (the last 64 bits from the 128 bits trace id)
  */
-fun Span.traceIdAsHexString(): String {
-    return context().toTraceId().toLong().toHexString()
+fun Span.leastSignificant64BitsTraceId(): String {
+    return (this as? DDSpan)?.traceId?.toString(16)?.padStart(32, '0')?.takeLast(16) ?: ""
```

Comment:
will we be aware about the cases when we get empty string? nothing will crash?

Reply:
based on the code we have there is no way we could get empty strings here as the generator is tested above and other tests will fail first. Here the debate is either we add these logs just for the sake of it or we just don't and consider this safe.

Reply:
also we will be aware as I added this extra checks:
```
  hasValidMostSignificant64BitsTraceId()
  hasValidLeastSignificant64BitsTraceId()
```
They validate the regEx